File: Samples CSharp/MultiQuery/MultiQueryForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using Therefore.API;

namespace MultiQuery
{
    public partial class MultiQueryForm : Form
    {
        public MultiQueryForm()
        {
            InitializeComponent();
        }

        private TheServer server;
        private TheMultiQueryResult multiResult;

        private void MultiQueryForm_Load(object sender, EventArgs e)
        {
            try
            {
	            // 1. Connect to the Therefore server
	            server = new TheServer();
	            server.Connect(TheClientType.CustomApplication);
            }
            catch (System.Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }

        private void btn_execute_Click(object sender, EventArgs e)
        {
            try
            {
	            // 2. Create a new multi-query
	            TheMultiQuery multiQuery = new TheMultiQuery();
	
	            // 3. Create and add two or more queries
	            TheQuery query = new TheQuery();
	            query.Category.Load("Files", server);
	            query.SelectFields.AddAll();
	            query.Conditions.Add("Filename", "*Moya*"); // using the column 
	            query.OrderByFields.Add("Creation_Date");
	            multiQuery.Add(query);
	
	            TheQuery query2 = new TheQuery();
	            query2.Category.Load("Simple Invoice", server);
	            query2.SelectFields.AddAll();
	            multiQuery.Add(query2);
	
	            // 4. Execute the multi-query
	            multiResult = multiQuery.Execute(server);
	
	            // Optional 5. Display multi-query definition and results in MessageBox
	            System.Windows.Forms.MessageBox.Show(multiQuery.ToString()
	                + "\r\n---------------------------------------------------\r\n\r\n"
	                + multiResult.ToString());
            }
            catch (System.Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }

        private void btn_process_Click(object sender, EventArgs e)
        {
            try
            {
	            txt_result.Text = "";
	            // Iterate through search results using nested for-loops
	            for (int i = 0; i < multiResult.Count; i++)
	            {
	                for (int j = 0; j < multiResult[i].Count; j++)
	                {
	                    for (int k = 0; k < multiResult[i][j].Count; k++)
	                    {
	                        // Access the k-th field in the j-th row of the i-th query
	                        if (this.multiResult[i][j][k] != null)
	                            txt_result.Text += multiResult[i][j][k].ToString() + "\r\n";
	                    }
	                }
	            }
	         
	            // Iterate through search results using the nested for-each constructs
	            foreach (TheQueryResult QueryResult in multiResult)
	            {
	                foreach (TheQueryResultRow row in QueryResult)
	                {
	                    foreach (object obj in row)
	                    {
	                        if (obj is DBNull)
	                            txt_result.Text += "null\r\n";
	                        else
	                            txt_result.Text += obj.ToString()+ "\r\n";
	                    }
	                }
	            }
            }
            catch (System.Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }
    }
}

File: Samples VB.NET/MultiQuery/MultiQueryForm.vb
Imports System.ComponentModel
Imports System.Text
Imports Therefore.API

Namespace MultiQuery
	Partial Public Class MultiQueryForm
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private server As TheServer
		Private multiResult As TheMultiQueryResult

		Private Sub MultiQueryForm_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Try
				' 1. Connect to the Therefore server
				server = New TheServer()
				server.Connect(TheClientType.CustomApplication)
			Catch ex As Exception
				MessageBox.Show(ex.Message)
			End Try
		End Sub

		Private Sub btn_execute_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btn_execute.Click
			Try
				' 2. Create a new multi-query
				Dim multiQuery As New TheMultiQuery()

				' 3. Create and add two or more queries
				Dim query As New TheQuery()
				query.Category.Load("Files", server)
				query.SelectFields.AddAll()
				query.Conditions.Add("Filename", "*Moya*") ' using the column
				query.OrderByFields.Add("Creation_Date")
				multiQuery.Add(query)

				Dim query2 As New TheQuery()
				query2.Category.Load("Simple Invoice", server)
				query2.SelectFields.AddAll()
				multiQuery.Add(query2)

				' 4. Execute the multi-query
				multiResult = multiQuery.Execute(server)

				' Optional 5. Display multi-query definition and results in MessageBox
				MessageBox.Show(CType(multiQuery, Object).ToString() & vbCrLf & "---------------------------------------------------" & vbCrLf & vbCrLf & CType(multiResult, Object).ToString())
			Catch ex As Exception
				MessageBox.Show(ex.Message)
			End Try
		End Sub

		Private Sub btn_process_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btn_process.Click
			Try
				txt_result.Text = ""
				' Iterate through search results using nested for-loops
				For i As Integer = 0 To multiResult.Count - 1
					For j As Integer = 0 To multiResult(i).Count - 1
						For k As Integer = 0 To multiResult(i)(j).Count - 1
							' Access the k-th field in the j-th row of the i-th query
							If Me.multiResult(i)(j)(k) IsNot Nothing Then
								txt_result.Text += multiResult(i)(j)(k).ToString() & vbCrLf
							End If
						Next k
					Next j
				Next i

				' Iterate through search results using the nested for-each constructs
				For Each QueryResult As TheQueryResult In multiResult
					For Each row As TheQueryResultRow In QueryResult
						For Each obj As Object In row
							If TypeOf obj Is DBNull Then
								txt_result.Text &= "null" & vbCrLf
							Else
								txt_result.Text += obj.ToString() & vbCrLf
							End If
						Next obj
					Next row
				Next QueryResult
			Catch ex As Exception
				MessageBox.Show(ex.Message)
			End Try
		End Sub
	End Class
End Namespace
